diff --git a/configs/example/fs.py b/configs/example/fs.py
index 3997ed76c..43bebcd66 100644
--- a/configs/example/fs.py
+++ b/configs/example/fs.py
@@ -376,5 +376,7 @@ if buildEnv['TARGET_ISA'] == "arm" and options.generate_dtb:
             sys = getattr(root, sysname)
             sys.dtb_filename = create_dtb_for_system(sys, '%s.dtb' % sysname)
 
+from m5.objects import ArmSemihosting
+test_sys.semihosting = ArmSemihosting()
 Simulation.setWorkCountOptions(test_sys, options)
 Simulation.run(options, root, test_sys, FutureClass)
